polling_stations/apps/file_uploads/utils.py

```diff
@@ -12,17 +13,43 @@ def get_domain(request):
     return os.environ.get("APP_DOMAIN", request.META.get("HTTP_HOST"))
 
 
+# This is a quick fix: Peter can raise a
+# PR to update CUSTOM_DOMAINS with the the actual
+# list of email domains and councils as the
+# need arises.
+def approved_custom_domain(user, email_domain):
+    # if email_domain in custom_domains, assign it to the
+    # corresponding council and return True
+    if email_domain not in CUSTOM_DOMAINS:
+        return False
+    else:
+        # get an instance of the council described in the custom_domains dict
+        council = Council.objects.using(LOGGER_DB_NAME).get(
+            council_id=CUSTOM_DOMAINS[email_domain]
+        )
+        UserCouncils.objects.using(LOGGER_DB_NAME).update_or_create(
+            user=user, council=council
+        )
+        return True
+
+
 def assign_councils_to_user(user):
     """
     Adds rows to the join table between User and Council
     """
     email_domain = user.email.rsplit("@", 1)[1]
-    councils = Council.objects.using(LOGGER_DB_NAME).filter(
-        Q(electoral_services_email__contains=email_domain)
-        | Q(registration_email__contains=email_domain)
-    )
-
-    for council in councils:
-        UserCouncils.objects.using(LOGGER_DB_NAME).update_or_create(
-            user=user, council=council
+    if approved_custom_domain(user, email_domain):
```

Comment:
I _think_ we might want to call both methods in all cases…

If someone at council A manages two areas, we want them to be assigned to two councils when they log in.

We can add `council-A.gov.uk` to your list to assign it to "Council B", but we also want the normal code to assign them to "Council A".

So I think this can just be:
```
assign_approved_custom_domain()
councils = Council.objects…
[etc]
```

Reply:
Fixed with 9c15a6cd5cf7a4948cfa5c3c331209fe1a0017c2